src/cmake-tools.ts

```diff
@@ -343,32 +377,41 @@ export class CMakeTools implements vscode.Disposable, api.CMakeToolsAPI {
    * Implementation of `cmake.configure`
    */
   configure(extra_args: string[] = []) {
-    return this._doConfigure(async (consumer) => {
-      const drv = await this._cmakeDriver;
-      return drv.configure(extra_args, consumer);
+    return this._doConfigure(async(consumer) => {
+      const drv = await this.getCMakeDriverInstance();
+      if (drv) {
+        return drv.configure(extra_args, consumer);
+      } else {
+        return -1;
+      }
     });
   }
 
   /**
    * Implementation of `cmake.cleanConfigure()
    */
   cleanConfigure() {
-    return this._doConfigure(async (consumer) => {
-      const drv = await this._cmakeDriver;
-      return drv.cleanConfigure(consumer);
+    return this._doConfigure(async(consumer) => {
+      const drv = await this.getCMakeDriverInstance();
+      if (drv) {
+        return drv.cleanConfigure(consumer);
+      } else {
+        return -1;
+      }
     });
   }
 
   /**
    * Wraps pre/post configure logic around an actual configure function
    * @param cb The actual configure callback. Called to do the configure
    */
-  private async _doConfigure(cb: (consumer: diags.CMakeOutputConsumer) => Promise<number>): Promise<number> {
-    if (!this._kitManager.activeKit) {
+  private async _doConfigure(cb: (consumer: diags.CMakeOutputConsumer) => Promise<number>):
+      Promise<number> {
+    if (!this._kitManager.hasActiveKit) {
       log.debug('No kit selected yet. Asking for a Kit first.');
       await this.selectKit();
     }
-    if (!this._kitManager.activeKit) {
+    if (!this._kitManager.hasActiveKit) {
```

Comment:
You are checking twice against `this._kitManager.hasActiveKit`.

Reply:
This is correct.
After the first check the user can select a kit.
The second check is needed if the user did not select a kit.